corehq/apps/app_manager/management/commands/migrate_app_to_cmitfb.py

```diff
@@ -39,23 +46,54 @@ def handle(self, *args, **options):
 
     def migrate_app(self, app_id):
         app = Application.get(app_id)
-        if app.vellum_case_management:
+        migrate_usercase = should_migrate_usercase(app, self.migrate_usercase)
+        if app.vellum_case_management and not migrate_usercase:
             logger.info('already migrated app {}'.format(app_id))
             return
 
         modules = [m for m in app.modules if m.module_type == 'basic']
         for module in modules:
             forms = [f for f in module.forms if f.doc_type == 'Form']
             for form in forms:
+                preloads = []
                 preload = form.actions.case_preload.preload
                 if preload:
                     if form.requires == 'case':
-                        xform = XForm(form.source)
-                        xform.add_case_preloads(preload)
-                        save_xform(app, form, ET.tostring(xform.xml))
-                        form.case_references = {"load": {path: [case_property]
-                            for path, case_property in preload.iteritems()}}
+                        preloads.append({
+                            "hashtag": "#case/",
+                            "preloads": preload,
+                        })
                     form.actions.case_preload = PreloadAction()
+                usercase_preload = form.actions.usercase_preload.preload
+                if migrate_usercase and usercase_preload:
+                    preloads.append({
+                        "hashtag": "#user/",
+                        "preloads": usercase_preload,
+                        "case_id_xpath": SESSION_USERCASE_ID,
+                    })
+                    form.actions.usercase_preload = PreloadAction()
+                if preloads:
+                    migrate_preloads(app, form, preloads)
 
         app.vellum_case_management = True
         app.save()
+
+
+def migrate_preloads(app, form, preloads):
+    xform = XForm(form.source)
+    for kwargs in preloads:
+        hashtag = kwargs.pop("hashtag")
+        xform.add_case_preloads(**kwargs)
+        refs = {path: [hashtag + case_property]
+                for path, case_property in kwargs["preloads"].iteritems()}
+        if form.case_references:
+            form.case_references["load"].update(refs)
+        else:
+            form.case_references = {"load": refs}
+    save_xform(app, form, ET.tostring(xform.xml))
+
+
+def should_migrate_usercase(app, migrate_usercase):
```

Comment:
The feature flag is domain level, but this migration can occur on one app at a time. I think that means that if you run this command on an app whose domain does not have the feature flag then their user case properties will show up in the form builder, but they can also add more in the case management screen. This will likely only affect QA at the start, but I could also seeing it happen to internal projects or FM projects.

For the original migration we added a property on the app to describe if it was migrated or not, because we did a staged rollout. That's probably not necessary here and you could force migrating entire domains at once, but I don't think that this will work well as is